From 1459e3f219ef447c68d179af76a7e179282ae554 Mon Sep 17 00:00:00 2001
From: LuK1337 <priv.luk@gmail.com>
Date: Sun, 04 Oct 2026 10:13:04 +0200
Subject: [PATCH] Revert "Fix Use-After-Free in NXP NFC HAL timer teardown"

Sadly this AI generated fix for AI generated bug doesn't work properly
and every once in a while will just get stuck on something, logspam and
drain your battery...

This reverts commit ad16c5c95a8613cc7ef8890c243068633e8d2f83.

Change-Id: I19e2f5d701cb5c8fb7c2047b9b402696bf52457e
---

diff --git a/pn8x/halimpl/hal/phNxpNciHal.cc b/pn8x/halimpl/hal/phNxpNciHal.cc
index e3c39d9..a246e52 100644
--- a/pn8x/halimpl/hal/phNxpNciHal.cc
+++ b/pn8x/halimpl/hal/phNxpNciHal.cc
@@ -385,11 +385,8 @@
   if (status != NFCSTATUS_SUCCESS) {
     if (NFCSTATUS_SUCCESS != phNxpNciHal_fw_mw_ver_check()) {
       NXPLOG_NCIHAL_D("Chip Version Middleware Version mismatch!!!!");
-      intptr_t client_id = nxpncihal_ctrl.gDrvCfg.nClientId;
-      nxpncihal_ctrl.gDrvCfg.nClientId = 0;
       phOsalNfc_Timer_Cleanup();
       phTmlNfc_Shutdown_CleanUp();
-      phDal4Nfc_msgrelease(client_id);
       status = NFCSTATUS_FAILED;
     } else {
       NXPLOG_NCIHAL_E("FW download failed, Continue NFCC init");
@@ -2069,11 +2066,8 @@
           (pphTmlNfc_TransactCompletionCb_t)&phNxpNciHal_read_complete, NULL);
       if (status != NFCSTATUS_PENDING) {
         NXPLOG_NCIHAL_E("TML Read status error status = %x", status);
-        intptr_t client_id = nxpncihal_ctrl.gDrvCfg.nClientId;
-        nxpncihal_ctrl.gDrvCfg.nClientId = 0;
         phOsalNfc_Timer_Cleanup();
         phTmlNfc_Shutdown();
-        phDal4Nfc_msgrelease(client_id);
         status = NFCSTATUS_FAILED;
       }
       break;
@@ -2191,8 +2185,6 @@
     status = phTmlNfc_ReadAbort();
     status = phTmlNfc_WriteAbort();
 
-    intptr_t client_id = nxpncihal_ctrl.gDrvCfg.nClientId;
-    nxpncihal_ctrl.gDrvCfg.nClientId = 0;
     phOsalNfc_Timer_Cleanup();
 
     status = phTmlNfc_Shutdown();
@@ -2203,7 +2195,7 @@
 
     phTmlNfc_CleanUp();
 
-    phDal4Nfc_msgrelease(client_id);
+    phDal4Nfc_msgrelease(nxpncihal_ctrl.gDrvCfg.nClientId);
 
     memset(&nxpncihal_ctrl, 0x00, sizeof(nxpncihal_ctrl));
 
@@ -2243,8 +2235,6 @@
     status = phTmlNfc_ReadAbort();
     status = phTmlNfc_WriteAbort();
 
-    intptr_t client_id = nxpncihal_ctrl.gDrvCfg.nClientId;
-    nxpncihal_ctrl.gDrvCfg.nClientId = 0;
     phOsalNfc_Timer_Cleanup();
 
     status = phTmlNfc_Shutdown();
@@ -2255,7 +2245,7 @@
 
     phTmlNfc_CleanUp();
 
-    phDal4Nfc_msgrelease(client_id);
+    phDal4Nfc_msgrelease(nxpncihal_ctrl.gDrvCfg.nClientId);
 
     memset(&nxpncihal_ctrl, 0x00, sizeof(nxpncihal_ctrl));
 
diff --git a/snxxx/halimpl/hal/phNxpNciHal.cc b/snxxx/halimpl/hal/phNxpNciHal.cc
index 2df281b..c0a39b8 100644
--- a/snxxx/halimpl/hal/phNxpNciHal.cc
+++ b/snxxx/halimpl/hal/phNxpNciHal.cc
@@ -489,12 +489,9 @@
       fw_download_success = TRUE;
     } else if (status == NFCSTATUS_FW_CHECK_INTEGRITY_FAILED ||
                (phNxpNciHal_fw_mw_ver_check() != NFCSTATUS_SUCCESS)) {
-      intptr_t client_id = nxpncihal_ctrl.gDrvCfg.nClientId;
-      nxpncihal_ctrl.gDrvCfg.nClientId = 0;
       phOsalNfc_Timer_Cleanup();
       phNxpTempMgr::GetInstance().Reset();
       phTmlNfc_Shutdown_CleanUp();
-      phDal4Nfc_msgrelease(client_id);
       return NFCSTATUS_CMD_ABORTED;
     }
 
@@ -2569,8 +2566,6 @@
     status = phTmlNfc_ReadAbort();
     status = phTmlNfc_WriteAbort();
 
-    intptr_t client_id = nxpncihal_ctrl.gDrvCfg.nClientId;
-    nxpncihal_ctrl.gDrvCfg.nClientId = 0;
     phOsalNfc_Timer_Cleanup();
 
     status = phTmlNfc_Shutdown();
@@ -2582,7 +2577,7 @@
     phNxpTempMgr::GetInstance().Reset();
     phTmlNfc_CleanUp();
 
-    phDal4Nfc_msgrelease(client_id);
+    phDal4Nfc_msgrelease(nxpncihal_ctrl.gDrvCfg.nClientId);
 
     memset(&nxpncihal_ctrl, 0x00, sizeof(nxpncihal_ctrl));
 
@@ -2652,8 +2647,6 @@
     if (status != NFCSTATUS_SUCCESS) {
       NXPLOG_TML_E("phTmlNfc_ReadAbort Failed");
     }
-    intptr_t client_id = nxpncihal_ctrl.gDrvCfg.nClientId;
-    nxpncihal_ctrl.gDrvCfg.nClientId = 0;
     phOsalNfc_Timer_Cleanup();
 
     status = phTmlNfc_Shutdown();
@@ -2669,7 +2662,7 @@
     phNxpTempMgr::GetInstance().Reset();
     phTmlNfc_CleanUp();
 
-    phDal4Nfc_msgrelease(client_id);
+    phDal4Nfc_msgrelease(nxpncihal_ctrl.gDrvCfg.nClientId);
 
     memset(&nxpncihal_ctrl, 0x00, sizeof(nxpncihal_ctrl));
   }
